### PR TITLE
HBASE-25127 Enhance PerformanceEvaluation to profile meta replica performance. 

### PR DESCRIPTION
**Pull request by clarax**

### CODE REVIEW

**Comment by clarax** [comment]
Back port to branch 2 with added changed to open access to MetaAccessor.makeDeleteFromRegionInfo. This change was part of https://issues.apache.org/jira/browse/HBASE-24633 checked into master but not back ported.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   2m  5s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 38s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 58s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  13m 19s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   2m 47s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  39m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2655 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 5b575a394a89 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 4b9a14d9f3 |
| Max. process+thread count | 84 (vs. ulimit of 12500) |
| modules | C: hbase-client hbase-mapreduce U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 37s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 13s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 59s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 42s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 28s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  10m 56s |  hbase-mapreduce in the patch passed.  |
|  |   |  42m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2655 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 1118bc674b5c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 4b9a14d9f3 |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/testReport/ |
| Max. process+thread count | 3552 (vs. ulimit of 12500) |
| modules | C: hbase-client hbase-mapreduce U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 36s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 55s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 34s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 31s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  16m  0s |  hbase-mapreduce in the patch passed.  |
|  |   |  46m 39s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2655 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux f5887ba5ebb7 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 4b9a14d9f3 |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/testReport/ |
| Max. process+thread count | 3307 (vs. ulimit of 12500) |
| modules | C: hbase-client hbase-mapreduce U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2655/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java [hunk 1]
```diff
@@ -1314,7 +1314,7 @@ public static Put makePutFromRegionInfo(RegionInfo regionInfo, long ts) throws I
   /**
    * Generates and returns a Delete containing the region info for the catalog table
    */
-  private static Delete makeDeleteFromRegionInfo(RegionInfo regionInfo, long ts) {
+  public static Delete makeDeleteFromRegionInfo(RegionInfo regionInfo, long ts) {
```
Comment (saintstack) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java:1317]: Is it like this on master branch?

This seems extreme for a test? We have to do this?

Comment (clarax) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java:1317]: This was part of Duo's https://issues.apache.org/jira/browse/HBASE-24633, checked into master but not back ported.

Comment (saintstack) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java:1317]: I don't see this change over in https://github.com/apache/hbase/pull/2003/files

There are conflicts w/ this patch?

Thanks @clarax

Comment (clarax) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java:1317]: Not sure what that PR was not in this branch but there shouldn't be conflicts.

**Pull request merged**